zng/resolver/marshal.go

```diff
@@ -741,3 +765,100 @@ func typeNameOfValue(value interface{}) (string, error) {
 	}
 	return fmt.Sprintf("%s.%s", typ.PkgPath(), typ.Name()), nil
 }
+
+func (u *UnmarshalContext) lookupType(typ zng.Type) (reflect.Type, error) {
+	switch typ := typ.(type) {
+	case *zng.TypeAlias:
+		template := u.binder.lookup(typ.Name)
+		if template == nil {
+			// Ignore aliases for which there are no bindings.
+			// If an interface type being marshaled into doesn't
+			// have a binding, then a type mismatch will be caught
+			// by reflect when the Set() method is called on the
+			// value and the concrete value doesn't implement the
+			// interface.
+			return u.lookupType(typ.Type)
+		}
+		return template, nil
+	case *zng.TypeRecord:
+		return u.lookupTypeRecord(typ)
+	case *zng.TypeArray:
+		elemType, err := u.lookupType(typ.Type)
+		if err != nil {
+			return nil, err
+		}
+		return reflect.SliceOf(elemType), nil
+	case *zng.TypeSet:
+		elemType, err := u.lookupType(typ.Type)
+		if err != nil {
+			return nil, err
+		}
+		return reflect.SliceOf(elemType), nil
+	case *zng.TypeUnion, *zng.TypeEnum:
+		// For now just return nil here. The layer above will flag
+		// a type error.  At some point, we can create Go-native data structures
+		// in package zng for representing a union or enum as a standalone
+		// entity.  See issue #1853.
+		return nil, nil
+	case *zng.TypeMap:
+		KeyType, err := u.lookupType(typ.KeyType)
+		if err != nil {
+			return nil, err
+		}
+		valType, err := u.lookupType(typ.ValType)
+		if err != nil {
+			return nil, err
+		}
+		return reflect.MapOf(KeyType, valType), nil
+	default:
+		return u.lookupPrimitiveType(typ)
+	}
+}
+
+func (u *UnmarshalContext) lookupTypeRecord(typ *zng.TypeRecord) (reflect.Type, error) {
+	return nil, errors.New("unmarshaling records into interface value requires type binding")
+}
+
+func (u *UnmarshalContext) lookupPrimitiveType(typ zng.Type) (reflect.Type, error) {
+	var v interface{}
+	switch typ := typ.(type) {
+	// XXX We should have counterparts for bstring, error, and type type.
+	// See issue #1853.
+	case *zng.TypeOfString, *zng.TypeOfBstring, *zng.TypeOfError, *zng.TypeOfType:
+		v = ""
+	case *zng.TypeOfBool:
+		v = true
```

Comment:
Make this the zero value like the others.
```suggestion
		v = false
```